aten/src/ATen/native/native_functions.yaml

```diff
@@ -10079,12 +10079,12 @@
   python_module: linalg
   variants: function
   dispatch:
-    CompositeExplicitAutograd: linalg_eigh
+    CPU, CUDA: linalg_eigh
 
 - func: linalg_eigh.eigvals(Tensor self, str UPLO="L", *, Tensor(a!) eigvals, Tensor(b!) eigvecs) -> (Tensor(a!) eigenvalues, Tensor(b!) eigenvectors)
   python_module: linalg
   dispatch:
-    CompositeExplicitAutograd: linalg_eigh_out
+    CPU, CUDA: linalg_eigh_out
```

Comment:
`checkSameDevice` from linalg_eigh_out should also be removed, now that device checks are codegened. Same for other `checkSameDevice` callse (there are 45 in BatchLinearAlgebra.cpp)

Reply:
Removed `checkSameDevice` from `linalg_eigh_out` and `linalg_eigvalsh_out` here. Will do the same for everything else in a separate PR.